joern-cli/frontends/pysrc2cpg/src/main/scala/io/joern/pysrc2cpg/PythonTypeHintCallLinker.scala

```diff
@@ -18,4 +21,41 @@ class PythonTypeHintCallLinker(cpg: Cpg) extends XTypeHintCallLinker(cpg) {
     case typ => typ
   }
 
+  override def linkCalls(builder: DiffGraphBuilder): Unit = {
+    val methodMap = mutable.HashMap.empty[String, MethodBase]
+    val callerAndCallees = calls
+      .map(call => (call, calleeNames(call)))
+      .toList
+    // Gather all method nodes and/or stubs
+    callerAndCallees
+      .foreach { case (call, methodNames) =>
+        val ms = callees(methodNames).l
+        if (ms.nonEmpty) {
+          ms.foreach { m => methodMap.put(m.fullName, m) }
+        } else {
+          val mNames = ms.map(_.fullName).toSet
+          methodNames
+            .filterNot(mNames.contains)
+            .map(fullName => createMethodStub(fullName, call, builder))
+            .foreach { m => methodMap.put(m.fullName, m) }
+        }
+      }
+    // Link edges to method nodes
+    callerAndCallees.foreach { case (call, methodNames) =>
+      methodNames
+        .flatMap(methodMap.get)
+        .foreach { m => builder.addEdge(call, m, EdgeTypes.CALL) }
+      if (methodNames.size == 1) {
+        builder.setNodeProperty(call, PropertyNames.METHOD_FULL_NAME, methodNames.head)
+      } else if (methodNames.size > 1) {
+        val builtInMethodName = methodNames.filter(s => s.contains("__builtin"))
+        if (builtInMethodName.size > 0) {
+          builder.setNodeProperty(call, PropertyNames.METHOD_FULL_NAME, builtInMethodName.sortBy(_.length).head)
+        } else {
+          builder.setNodeProperty(call, PropertyNames.METHOD_FULL_NAME, methodNames.sortBy(_.length).head)
```

Comment:
Not sure if I'm a fan of simply taking the longest path. If there are a lot of `<indexAccess>` calls, then the longest one would be the wrong one. e.g.
```python
def bar():
  pass

bar()
db = [bar]
db.bar()
```
```
joern> cpg.call("bar").callee.fullName.l
res0 = List("Foo.py:<module>.bar", "Foo.py:<module>.db.<indexAccess>.bar")
```
In this case, both calls should be `Foo.py:<module>.bar`

Reply:
Maybe replace the length check with the check that filters out calls with dummy values

Reply:
Changes in this file are no longer needed after [this](https://github.com/joernio/joern/blob/437c4a73d46ed25bfe549861481ac9c6bf05e929/joern-cli/frontends/pysrc2cpg/src/main/scala/io/joern/pysrc2cpg/PythonAstVisitor.scala#L1690) change. Restored this code to what it is in master

Reply:
Unresolved this conversation since I brought this back due to UT failures. Will get back to this once `DYNAMIC_DISPATCH` related UT issues are sorted out

Reply:
Change the following from
```scala
      if (methodNames.size == 1) {
        builder.setNodeProperty(call, PropertyNames.METHOD_FULL_NAME, methodNames.head)
      } else if (methodNames.size > 1) {
        val methodNamesFiltered = methodNames.filter(!isDummyType(_))
        val builtInMethodName   = methodNamesFiltered.filter(s => s.contains("__builtin"))
        if (builtInMethodName.size > 0) {
          builder.setNodeProperty(call, PropertyNames.METHOD_FULL_NAME, builtInMethodName.sortBy(_.length).head)
        } else {
          builder.setNodeProperty(call, PropertyNames.METHOD_FULL_NAME, methodNames.sortBy(_.length).head)
        }
      }
```
to
```scala
      if (methodNames.size == 1) {
        builder.setNodeProperty(call, PropertyNames.METHOD_FULL_NAME, methodNames.head)
      } else if (methodNames.size > 1) {
        val nonDummyMethodNames = methodNames.filterNot(isDummyType)
        if (nonDummyMethodNames.nonEmpty) {
          builder.setNodeProperty(call, PropertyNames.METHOD_FULL_NAME, nonDummyMethodNames.minBy(_.length))
        }
      }
```
This is the only "sound" heuristic I would imagine from the above combinations. Simply because there is a built-in does not always mean it's the correct call over some otherwise.

Reply:
Done